Mage/src/main/java/mage/players/PlayerImpl.java

```diff
@@ -539,6 +539,7 @@ public Counters getCounters() {
     public void beginTurn(Game game) {
         resetLandsPlayed();
         updateRange(game);
+        game.getState().removeTurnStartEffect(game);
```

Comment:
When exactly this should be called during the .
It should be after the turn's skipping logic is done, but before anything might trigger off an old trigger.

Reply:
It's fine to call it here. But make sure it will support leaving players too:

> 800.4m. When a player leaves the game, any continuous effects with durations that last until that player's next turn or until a specific point in that turn will last until that turn would have begun. They neither expire immediately nor last indefinitely.

Looks like current code will be never called for leaved player and will keep trigger forever in game state. If it's true then player.isInGame will help with it:
![shot_240517_095638](https://github.com/magefree/mage/assets/8344157/bf14c21a-6827-4c23-8aa8-c28299f6d8b7)

Reply:
Thanks I did miss that.
I'll make test for it, and fix it in asap.

Reply:
So after testing, the DelayedTriggers get cleared once the player leave/concede.
Which is incorrect, but less annoying, as the trigger expires just not at the right time.

By the way, If there are delayed triggers of a left player to stack, who decides stacking order? I need to read more on that in the CR.

Reply:
> 800.4d. If an [object](https://yawgatog.com/resources/magic-rules/#object) that would be owned by a [player](https://yawgatog.com/resources/magic-rules/#player) who has left the game would be created in any [zone](https://yawgatog.com/resources/magic-rules/#zone), it isn't created. If a [triggered ability](https://yawgatog.com/resources/magic-rules/#triggered_ability) that would be controlled by a [player](https://yawgatog.com/resources/magic-rules/#player) who has left the game would be put onto the [stack](https://yawgatog.com/resources/magic-rules/#stack), it isn't put on the [stack](https://yawgatog.com/resources/magic-rules/#stack).

So triggers (which delayed triggers are) for leaving player don't trigger. So current no trigger after player left is correct behavior. I'll add proper cleanup though to not keep DelayedTriggerAbility the whole game.